logr.go

```diff
@@ -204,7 +240,8 @@ func FromContext(ctx context.Context) Logger {
 		return v
 	}
 
-	return nil
+	//FIXME: what to do here?  Could switch to pointers, but yuck?
```

Comment:
I would not switch to pointers, as it will just lead to a lot of unnecessary heap allocs.

Some options:

- Return the Discard logger, which is the same as `FromContextOrDiscard` is doing
- Add an error return value
- panic()

Note that this API change is breaking anyway, because the caller cannot compare a concrete struct against `nil`, so adding an error return value would not be too bad.

Reply:
agree